jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -1530,8 +1530,12 @@ ecma_op_function_try_to_lazy_instantiate_property (ecma_object_t *object_p, /**<
     bytecode_data_p = ecma_op_function_get_compiled_code ((ecma_extended_object_t *) object_p);
 
 #if ENABLED (JERRY_ESNEXT)
+    uint16_t function_type = CBC_FUNCTION_GET_TYPE (bytecode_data_p->status_flags);
+
     if (!(bytecode_data_p->status_flags & CBC_CODE_FLAGS_STRICT_MODE)
-        && !(CBC_FUNCTION_GET_TYPE (bytecode_data_p->status_flags) == CBC_FUNCTION_ARROW))
+        && !(function_type == CBC_FUNCTION_ARROW)
+        && !(function_type == CBC_FUNCTION_ASYNC)
+        && !(function_type == CBC_FUNCTION_GENERATOR))
```

Comment:
I checked accessors and constructor functions and it seems they don't have these properties as well. Is there any other function type which has these properties? Because if only normal functions have these properties, we can change the check to:
`CBC_FUNCTION_GET_TYPE (bytecode_data_p->status_flags) == CBC_FUNCTION_NORMAL`

Reply:
I think you are right. I checked the newest documentation and it looks like currently only normal function has these properties.